pysoa/common/transport/redis_gateway/core.py

```diff
@@ -347,7 +347,22 @@ def send_message(
             # Try at least once, up to queue_full_retries times, then error
             for i in range(-1, self.queue_full_retries):
                 if i >= 0:
-                    time.sleep((2 ** i + random.random()) / self.EXPONENTIAL_BACK_OFF_FACTOR)
+                    backoff_time_in_seconds = (2 ** i + random.random()) / self.EXPONENTIAL_BACK_OFF_FACTOR
```

Comment:
wonder what that random is for?

Reply:
You don't want all the clients retrying at the same time I guess. So the random part will help will that.